examples/colour-uniform/main.rs

```diff
@@ -804,11 +805,12 @@ impl<B: Backend> RenderPassState<B> {
             };
 
             let dependency = pass::SubpassDependency {
-                passes: pass::SubpassRef::External .. pass::SubpassRef::Pass(0),
-                stages: PipelineStage::COLOR_ATTACHMENT_OUTPUT
-                    .. PipelineStage::COLOR_ATTACHMENT_OUTPUT,
-                accesses: i::Access::empty()
-                    .. (i::Access::COLOR_ATTACHMENT_READ | i::Access::COLOR_ATTACHMENT_WRITE),
+                src_subpass: pass::SubpassRef::External,
```

Comment:
My biggest concern here is duplication of code. Would you be open to a new struct that encloses the duplicated types? Something like

```
struct DependencySide {
  subpass: SubRef,
  stage: PipelineStage,
  access: image::Access,
}
struct SubpassDependency {
   src: DependencySide,
   dst: DependencySide,
}
```

Reply:
Duplication of user code? Or duplication of gfx-hal code?

What are your thoughts on keeping the public API with `src_`/`dest_` fields and then using the DependencySide internally.

Reply:
duplication on the API side is what I mean